src/probability/probability_mass_function/constructions.lean

```diff
@@ -47,13 +47,22 @@ lemma mem_support_map_iff : b ∈ (map f p).support ↔ ∃ a ∈ p.support, f a
 
 lemma bind_pure_comp : bind p (pure ∘ f) = map f p := rfl
 
-lemma map_id : map id p = p := by simp [map]
+lemma map_id : map id p = p := bind_pure _
 
 lemma map_comp (g : β → γ) : (p.map f).map g = p.map (g ∘ f) :=
-by simp [map]
+by simp only [map, bind_bind, pure_bind]
```

Comment:
This is better not to change. The simp is quick enough and more robust to library changes.